flash-patcher.py

```diff
@@ -160,55 +193,153 @@ def inject(self):
                 split_line = line_stripped.split()
 
                 # Handle internal commands
-                if split_line[0] == "//" and split_line[1] == "cmd:":
-                    if split_line[2] == "skip":
-                        try:
-                            file_line_no += int(split_line[3])
-                            continue
-                        except ValueError:
-                            perror("")
-                            perror("Invalid skip amount: " + split_line[3])
-                            perror("Expected integer (got type \"str\")")
-                            perror("Aborting...")
-                            exit(1)
+                if split_line[:2] == ["//", "cmd:", "skip"]:
+                    n_str = split_line[3]
+                    try:
+                        n = int(n_str)
+                        continue
+                    except ValueError:
+                        exception(
+                            "Invalid skip amount: ",
+                            n,
+                            "\n",
+                            "Expected integer.\n",
+                            "Aborting...",
+                        )
+                        sys.exit(1)
+
+                    file_line_no += n
 
                 self.fileContents[file.fileName].insert(file_line_no, line)
 
                 patch_line_no += 1
                 file_line_no += 1
-            
-        for file, fileContent in self.fileContents.items():
-            write_to_file(file, fileContent)
 
-"""
-Apply a single patch file.
-patch_file parameter: The path to the patch file.
-"""
-def apply_patch(patch_file):
+        for file, file_content in self.fileContents.items():
+            write_to_file(file, file_content)
+
+
+def check_jpexs_exists(path: str) -> bool:
+    """Check if JPEXS exists at a given path.
+
+    Returns True if successful, and False otherwise.
+    """
+    return path.exists()
+
+
+def detect_jpexs() -> JPEXSInterface:
+    """Detect and record JPEXS install location.
+
+    Checks various common JPEXS paths
+    Returns True if successful.
+    """
+    # apt install location
+    if check_jpexs_exists(LOCATION_APT):
+        return JPEXSInterface(LOCATION_APT)
+
+    # flatpak install location
+    if check_jpexs_exists(LOCATION_FLATPAK):
+        # Detect if JPEXS is installed.
+        # The function call can only detect if Flatpak is installed.
```

Comment:
nit: better comment
"if Flatpak is installed, `path.exists(flatpak)` will be true
if JPEXS is installed via flatpak, `flatpak run [a bunch of args] jpexs -h` will return successfully"